### PR TITLE
feat: Base class for directives and components

### PR DESCRIPTION
**Pull request by satanTime**

### CODE REVIEW

File: lib/mock-pipe/mock-pipe.ts [hunk 1]
```diff
@@ -3,20 +3,37 @@ import { Pipe, PipeTransform, Type } from '@angular/core';
...
 
 const defaultTransform = (...args: any[]): void => undefined;
-export function MockPipe<TPipe extends PipeTransform>(pipe: Type<TPipe>,
-                                                      transform: TPipe['transform'] = defaultTransform): Type<TPipe> {
-  const pipeName = pipeResolver.resolve(pipe).name;
+export function MockPipe<TPipe extends PipeTransform>(
+  pipe: Type<TPipe>,
+  transform: TPipe['transform'] = defaultTransform,
+): Type<MockedPipe<TPipe>> {
+  const cacheHit = cache.get(pipe);
```
Comment (satanTime) [inline lib/mock-pipe/mock-pipe.ts]: is it okay to cache pipe? It wasn't cached, and I assume there's a reason for that.

Comment (ike18t) [reply lib/mock-pipe/mock-pipe.ts]: They were cached in the past but I noticed an issue when I tried to declare a new implementation for a pipe in another test file (it was around translations and in one case I cared and in another i didn't) where it didn't let me.

Comment (satanTime) [reply lib/mock-pipe/mock-pipe.ts]: Aha, I see, good, then I'll write tests for it and remove the pipe cache from the PR.

**Comment by kekel87** [comment]
It sounds great to me. Great work! It's much clearer like that.

**Comment by satanTime** [comment]
@kekel87 @ike18t, please review, PR has been updated.

**Comment by satanTime** [comment]
Hi @kekel87, have you got a chance to review whether it solves your issue?

**Comment by kekel87** [comment]
@satanTime I just tested, it's good for me. Great work ! :+1:

**Pull request merged**